File: 118.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> vvi;
        vector<int> vi;
        for (int i=0;i<numRows;i++) {
            vector<int> v;
            if (i==0) {
                vi.push_back(1);
                vvi.push_back(vi);
                continue;
            }
            v=vi;
            for (int j=1;j<v.size();j++)
                v[j]=vi[j]+vi[j-1];
            v.push_back(1);
            vi=v;
            vvi.push_back(v);
        }
        return vvi;
    }
};

File: 118.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        if numRows==0:
            return []
        c=[]
        a=[1]
        c.append(a)
        for i in range(1,numRows):
            b=[1]
            for j in range(1,i):
                b.append(a[j-1]+a[j])
            b.append(1)
            a=b
            c.append(a)
        return c
